draft-sullivan-tls-implicit-ech.md

```diff
@@ -148,10 +193,35 @@ Other aspects of the base ECH spec remain unchanged. In particular, the client
 still picks a cipher suite from key_config.cipher_suites, produces a valid HPKE
 ephemeral key, and encrypts ClientHelloInner into the payload field.
 
-If the client-facing server issues an ECH retry hint (for example, in
-EncryptedExtensions), the client MUST still confirm that the server certificate
-is valid for the public_name from the ECHConfig used to establish the connection.
-Note that this may be a different name than the one sent in the outer SNI.
+If ECH is accepted, then the same procedure as-in draft-ietf-tls-esni is followed.
+However, is ECH is rejected and the Implict ECH was advertised, the client must follow
+a different strategy detailed in the next section.
+
+## Behavior when ECH fails
+
+Firstly, the client looks at the server's EncryptedExtensions message and checks for the presence
+of an ECH extension with retry_configs. If present, the client extracts any provided ECHConfigs
+which have the ImplicitECHAuthenticator extension and ignores any without it.
+
+If the extension is present, the client must hash the provided PublicKey using indicated hash algorithm and check that it appears in the HashedPublicKey list provided in the initialImplicitECH extension. If so,
```

Comment:
It is not 100% clear that this refers to the extension that the client received via other means (either a DNS SVCB lookup or a previous retry config.  There's a potential for this to be mistaken to refer to the value in the retry config, which would not work.